src/npc.cpp

```diff
@@ -1455,7 +1455,15 @@ void npc::decide_needs()
         elem = 20;
     }
     if( weapon.is_gun() ) {
-        needrank[need_ammo] = 5 * get_ammo( ammotype( *weapon.type->gun->ammo.begin() ) ).size();
+        int ups_drain = weapon.get_gun_ups_drain();
+        if( ups_drain > 0 ) {
+            int ups_charges = has_charges( "UPS_off", ups_drain ) + 
```

Comment:
I don't get the idea.

`has_charges` returns Boolean. You either get sum of two `true` or sum of two `false`, so basically either 2 or 0.

Reply:
Doh!  I thought it turn the amount.  I want charges_of(), don't I?

Reply:
Yes, looking into doxygen it should be it.

Reply:
Fixed, thanks.